OverlayPlugin.Core/MemoryProcessors/ContentFinderSettings/LineContentFinderSettings.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Advanced_Combat_Tracker;
+
+namespace RainbowMage.OverlayPlugin.MemoryProcessors.ContentFinderSettings
+{
+    class LineContentFinderSettings
+    {
+        public const uint LogFileLineID = 265;
+        private ILogger logger;
+        private readonly FFXIVRepository ffxiv;
+
+        private Func<string, DateTime, bool> logWriter;
+
+        private IContentFinderSettingsMemory contentFinderSettingsMemory;
+
+        private bool wroteFirstLine = false;
+
+        public LineContentFinderSettings(TinyIoCContainer container)
+        {
+            logger = container.Resolve<ILogger>();
+            ffxiv = container.Resolve<FFXIVRepository>();
+            if (!ffxiv.IsFFXIVPluginPresent())
+                return;
+            contentFinderSettingsMemory = container.Resolve<IContentFinderSettingsMemory>();
+            var customLogLines = container.Resolve<FFXIVCustomLogLines>();
+            this.logWriter = customLogLines.RegisterCustomLogLine(new LogLineRegistryEntry()
+            {
+                Name = "ContentFinderSettings",
+                Source = "OverlayPlugin",
+                ID = LogFileLineID,
+                Version = 1,
+            });
+            ActGlobals.oFormActMain.BeforeLogLineRead += LogLineHandler;
+        }
+
+        private void LogLineHandler(bool isImport, LogLineEventArgs args)
+        {
+            if (isImport)
+            {
+                return;
+            }
+
+            if (!contentFinderSettingsMemory.IsValid())
+                return;
+
+            try
+            {
+                LogMessageType lineType = (LogMessageType)args.detectedType;
+
+                if (lineType != LogMessageType.ChangeZone && wroteFirstLine)
+                    return;
+                wroteFirstLine = true;
+                WriteInContentFinderSettingsLine(args.detectedTime);
+            }
+            catch (Exception e)
+            {
+                logger.Log(LogLevel.Error, "Failed to process log line: " + e.ToString());
+            }
+        }
+
+        private void WriteInContentFinderSettingsLine(DateTime dateTime)
+        {
+            var inContentFinderContent = contentFinderSettingsMemory.GetInContentFinderContent();
+
+            // If we're not in a content finder content instance, set this to null
+            // So that we can default to all 0's later
+            var settings = inContentFinderContent ? contentFinderSettingsMemory.GetContentFinderSettings() : null;
```

Comment:
I'm a bit torn on this because on one hand I'd rather expose each memory read logic path as its own function, but now with actual synced ilvl that's no longer possible.

Reply:
I am happy with whatever you think makes sense. I think mostly I was imagining an internal memory read logic path as its own separate function, but if these settings are never valid when you're not in content finder content then it made some sense to me to make sure other users didn't call that function and expecting something useful from it too.

Reply:
I hope the adjustment in latest commit makes sense?

For context, the original (flawed) implementation that never saw an actual commit/push was looking at another location for the actual settings flags (which also contained `lootRules` and `limitedLevelingRoulette`, which are why they're mentioned in the interface `TODO`), and this other location was always valid regardless of if you were in an actual CF instance or not. So it made more sense to have discrete code paths at that point.